change xz to gz

**Describe the bug**
[mise](https://github.com/jdx/mise) or asdf is requesting a file that doesn't exist
see the [cloudflare-go releases](https://github.com/cloudflare/cloudflare-go/releases) page for available assets

**Steps to reproduce**

```
mise use flarectl@0.85.0 --verbose
[DEBUG] ARGS: /home/REDACTED/.local/bin/mise use flarectl@0.85.0 --verbose
[DEBUG] Config {
    Config Files: [
        "~/.config/mise/config.toml",
    ],
}
[DEBUG] Toolset: flarectl@0.86.0
downloading
~/.local/share/mise/plugins/flarectl/bin/download
[DEBUG] $ ~/.local/share/mise/plugins/flarectl/bin/download
* Downloading flarectl release 0.85.0...
curl: (22) The requested URL returned error: 404
asdf-flarectl: Could not download https://github.com/cloudflare/cloudflare-go/releases/download/v0.85.0/flarectl_0.85.0_linux_amd64.tar.xz
[ERROR] ~/.local/share/mise/plugins/flarectl/bin/download failed
Error:
   0: ~/.local/share/mise/plugins/flarectl/bin/download exited with non-zero status: exit code 1

Location:
   /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/core/src/convert/mod.rs:757

Version:
   2024.1.30 linux-x64 (3a195bc 2024-01-27)

Suggestion: Run with --verbose or MISE_VERBOSE=1 for more information.

Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.
```

**Expected behavior**

flarectl should successfully install
mise or asdf should retrieve the `gz` file

```
```

--- a/bin/download
+++ b/bin/download
@@ -10,10 +10,9 @@ source "${plugin_dir}/lib/utils.bash"
 
 mkdir -p "$ASDF_DOWNLOAD_PATH"
 
-# TODO: Adapt this to proper extension and adapt extracting strategy.
-release_file="$ASDF_DOWNLOAD_PATH/$TOOL_NAME-$ASDF_INSTALL_VERSION.tar.xz"
+release_file="$ASDF_DOWNLOAD_PATH/$TOOL_NAME-$ASDF_INSTALL_VERSION.tar.gz"
 
-# Download tar.xz file to the download directory
+# Download tar.gz file to the download directory
 download_release "$ASDF_INSTALL_VERSION" "$release_file"
 
 tar -xf "$release_file" -C "$ASDF_DOWNLOAD_PATH" || fail "Could not extract $release_file"

--- a/lib/utils.bash
+++ b/lib/utils.bash
@@ -53,7 +53,7 @@ download_release() {
     local arch=$arch_test
   fi
 
-  url="$GH_REPO/releases/download/v${version}/flarectl_${version}_${platform}_${arch}.tar.xz"
+  url="$GH_REPO/releases/download/v${version}/flarectl_${version}_${platform}_${arch}.tar.gz"
 
   echo "* Downloading $TOOL_NAME release $version..."
   curl "${curl_opts[@]}" -o "$filename" -C - "$url" || fail "Could not download $url"

